instructor/common.py

```diff
@@ -1728,20 +1740,59 @@ def facs_histograms_edit(request, assignment, sample_prep, sp):
 def select_images(request):
     pk = request.session['assignment_id']
     mapping_id = request.POST.get('mapping_pk')
+    filter_group_id = request.POST.get('filter_group_id')
     image_pk_list = request.POST.getlist('image_pk_list[]')
-    instance = get_object_or_404(
-        models.MicroscopyImageMapping,
-        pk=mapping_id,
-        sample_prep__assignment__pk=pk
-    )
-    for image_pk in image_pk_list:
+
+    if filter_group_id:
+        filter_group_id = filter_group_id.split('-')
+        filter_name = filter_group_id[0]
+        group_id = filter_group_id[1]
+        image_group = get_object_or_404(
+            models.MicroscopyGroupedImages,
+            id=group_id,
+            image_mapping__sample_prep__assignment__pk=pk
+        )
         selected_image = get_object_or_404(
             models.MicroscopyImage,
-            pk=image_pk,
+            pk=image_pk_list[0],
             assignment__pk=pk
         )
+        field_name = filter_name + '_filter_image'
+        setattr(image_group, field_name, selected_image)
```

Comment:
It's generally bad practice to use `setattr` or `getattr` or other reflection-style methods with user input, since it gives the user a chance to change other parts of the `image_group` object depending on what they passed in for `field_name`.